src/frontend/apps/desk/src/features/mail-domains/components/forms/CreateMailboxForm.tsx

```diff
@@ -0,0 +1,317 @@
+
+  const onSubmitCallback = (event: React.FormEvent) => {
+    event.preventDefault();
+    void methods.handleSubmit((data) =>
+      createMailbox({ ...data, mailDomainId: mailDomain.id }),
+    )();
+  };
+
+  return (
+    <FormProvider {...methods}>
+      <Modal
+        isOpen
+        leftActions={
+          <Button
+            color="secondary"
+            fullWidth
+            onClick={closeModal}
+            disabled={methods.formState.isSubmitting}
+          >
+            {t('Cancel')}
+          </Button>
+        }
+        onClose={closeModal}
+        closeOnClickOutside
+        hideCloseButton
+        rightActions={
+          <Button
+            color="primary"
+            fullWidth
+            type="submit"
+            form={FORM_ID}
+            disabled={methods.formState.isSubmitting}
+          >
+            {t('Submit')}
+          </Button>
+        }
+        size={ModalSize.LARGE}
+        title={
+          <Box $align="center" $gap="1rem">
+            <IconCreateMailbox
+              width={48}
+              color={colorsTokens()['primary-text']}
+              title={t('Mailbox creation form')}
+            />
+            <Text
+              $size="h3"
+              $margin={{ all: 'none' }}
+              role="heading"
+              aria-level={3}
+            >
+              {t('Create a mailbox')}
+            </Text>
+          </Box>
+        }
+      >
+        <ModalStyle />
+        <Box $width="100%" $margin={{ top: 'large', bottom: 'xl' }}>
+          {queryState.isError && (
+            <TextErrors className="mb-s" causes={queryState.error.cause} />
+          )}
+          {methods ? (
+            <Form
+              methods={methods}
+              t={t}
+              mailDomain={mailDomain}
+              onSubmitCallback={onSubmitCallback}
+            />
+          ) : null}
+        </Box>
+      </Modal>
+    </FormProvider>
+  );
+};
+
+const Form = ({
+  methods,
+  t,
+  mailDomain,
+  onSubmitCallback,
+}: {
+  methods: UseFormReturn<CreateMailboxParams>;
+  t: TFunction;
+  mailDomain: MailDomain;
+  onSubmitCallback: (event: React.FormEvent) => void;
+}) => (
+  <form onSubmit={onSubmitCallback} id={FORM_ID}>
+    <Box
+      $direction="column"
+      $width="100%"
+      $gap="3rem"
+      $margin={{ all: 'auto' }}
+    >
+      <FieldGroup>
+        <Field>
+          <Controller
+            control={methods.control}
+            name="first_name"
+            render={({ field, fieldState }) => (
+              <Input
+                aria-invalid={!!fieldState.error}
+                label={t('First name')}
+                state={fieldState.error ? 'error' : 'default'}
+                text={fieldState.error?.message}
+                onBlur={field.onBlur}
+                onChange={(e) => methods.setValue(field.name, e.target.value)}
+                value={String(field.value)}
+              />
```

Comment:
What about this way, like that after an error if the user do a change, the error will not be displayed anymore ?
What about personalized error message ?

```suggestion
              <Input
                aria-invalid={!!fieldState.error}
                label={t('First name')}
                state={fieldState.error ? 'error' : 'default'}
                text={
                  fieldState.error
                    ? t('Please enter your first name')
                    : undefined
                }
                {...methods.register('first_name')}
              />

```